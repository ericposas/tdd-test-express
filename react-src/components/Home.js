import React, { Component } from 'react'
import useName from './hooks/useName'

function Home({ ...props }) {

  const [content, setContent] = React.useState('')

  const name = useName('Boff')

  React.useEffect(() => {
    setContent(props.name)
  }, [props.name])

  return (
    <>
      <div>
        <div>
          {name}
        </div>
        <br/>
      </div>
    </>
  )
}

export default Home
