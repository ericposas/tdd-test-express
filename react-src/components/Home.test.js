import React, { Component } from 'react'
import { shallow } from 'enzyme'
import Home from './Home'

beforeEach(() => {
  jest.spyOn(React, 'useEffect').mockImplementationOnce(f => f())
})

test('component should have text content', () => {
  const text = 'homie.'
  const wrapper = shallow(<Home name={text}/>)
  expect(wrapper.text()).toEqual(text)
  expect(wrapper.containsMatchingElement(<div>{text}</div>)).toBeTruthy()
  expect(wrapper.containsMatchingElement(<br/>)).toBeTruthy()
  expect(wrapper).toMatchSnapshot()
  wrapper.unmount()
})
