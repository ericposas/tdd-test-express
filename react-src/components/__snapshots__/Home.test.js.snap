// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component should have text content 1`] = `
<Fragment>
  <div>
    <div>
      homie.
    </div>
    <br />
  </div>
</Fragment>
`;
